cfpb/cfgov-refresh
# Data migrations to convert Image & Text and Link Blob Groups to Info Unit Groups

[Info Unit Groups came on the scene quite a while ago]() and we are (I am) long overdue for migrating the data out of the blocks they replaced. This PR begins that work.

## Additions

- Data migration for Image & Text Groups to Info Unit Groups
- Data Migration for Link Blob Groups to Info Unit Groups

## Changes

-  Modify `get_stream_data` migration utility to handle corrupt revisions
  - Specifically, handle the case where a revision is missing the StreamField named `field_name`.

## Testing

1. Pull branch
1. Refresh your database (which runs the migrations)
1. Look at a wide range of pages to confirm that data has transferred as expected (use the Block Inventory on Content to look for pages to check)
1. Update your local Block Inventory with `./cfgov/manage.py block_inventory` and see that your local Block Inventory menu no longer offers to search for the following:
   - `v1.atomic_elements.molecules.HalfWidthLinkBlob` – It's still there! See note below.
   - `v1.atomic_elements.molecules.ImageText2575`
   - `v1.atomic_elements.molecules.ImageText5050`
   - `v1.atomic_elements.molecules.ThirdWidthLinkBlob`
   - `v1.atomic_elements.organisms.HalfWidthLinkBlobGroup`
   - `v1.atomic_elements.organisms.ImageText2575Group`
   - `v1.atomic_elements.organisms.ImageText5050Group`
   - `v1.atomic_elements.organisms.ThirdWidthLinkBlobGroup`

## Notes

- One instance of `HalfWidthLinkBlob`s remains – the weird special case on the `HomePage`'s `header` StreamField. We'll take care of that in a future PR.

## Todos

In this PR:
- [x] Write tests for the migration functions
- [x] Remove the organism classes

In future PRs:
- [ ] Remove the molecule classes
- [ ] Convert the singleton Half Width Link Blob molecules on the `HomePage` to Info Units
- [ ] Remove the old Info Unit and Info Unit Group templates

## Checklist
